src/audio/module_adapter/module/waves/waves.c

```diff
@@ -353,6 +354,7 @@ static int waves_effect_init(struct processing_module *mod)
 		comp_err(dev, "waves_effect_init() MaxxEffect_Initialize returned %d", status);
 		return -EINVAL;
 	}
+	waves_codec->initialized = true;
```

Comment:
does it need to set back to false by waves_codec_reset()?

Reply:
1. the sequence of .prepare(which calls MaxxEffect_Initialize()), .set_configuration(which calls waves_effect_message()) and .reset (which calls MaxxEffect_Reset()) is correct.
2. This patch aims to prevent CRAS from sending a blob while it is initializing. In this case, the MaxxEffect_Initialize() will not be called.
3. It is good practice to use the waves_codec->initialized variable to ensure symmetry between initialization and reset.

Reply:
I am curious of the hazard before the fix. Could this bug bring about DSP error?
It is a common case that set_configuration() may be called before prepare(). I am sure we have been done that way to Waves in previous projects, but we didn't feel something wrong.